### PR TITLE
bpo-32893: ast.literal_eval() no longer accepts booleans as numbers in AST.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue32893

### CODE REVIEW

**Review by auvipy — APPROVED** [review]

File: Lib/ast.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def _convert(node):
         elif isinstance(node, BinOp) and isinstance(node.op, (Add, Sub)):
             left = _convert_signed_num(node.left)
             right = _convert_num(node.right)
-            if isinstance(left, (int, float)) and isinstance(right, complex):
+            if type(left) in (int, float) and type(right) is complex:
```
Comment (isidentical) [inline Lib/ast.py:83]: I don't think it affects much but looks better to me
```suggestion
            if type(left) in {int, float} and type(right) is complex:
```

File: Misc/NEWS.d/next/Library/2018-02-21-11-59-21.bpo-32893.bLWZNi.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ast.literal_eval() no longer accepts booleans as numbers when pass AST
```
Comment (isidentical) [inline Misc/NEWS.d/next/Library/2018-02-21-11-59-21.bpo-32893.bLWZNi.rst:1]: Is there a need for `when pass AST argument.` part? You can give argument as string (like `ast.literal_eval("True+6j")` and still get `ValueError`.

File: Lib/ast.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/ast.py:83]: It is faster to create a tuple than a set.

File: Misc/NEWS.d/next/Library/2018-02-21-11-59-21.bpo-32893.bLWZNi.rst (hunk 2 above)
Comment (serhiy-storchaka) [reply Misc/NEWS.d/next/Library/2018-02-21-11-59-21.bpo-32893.bLWZNi.rst:1]: You get ValueError in `ast.literal_eval("True+6j")`. This PR only changes the behavior of `ast.literal_eval()` when pass AST argument.

Comment (isidentical) [reply Misc/NEWS.d/next/Library/2018-02-21-11-59-21.bpo-32893.bLWZNi.rst:1]: Ah,  I see.

**Review by isidentical — APPROVED** [review]

**Comment by isidentical** [comment]
@serhiy-storchaka does this PR still relevant to master (and 3.8)? I couldn't reproduce the examples you gave on tracker or the tests here. They all raise error as intended.

**Comment by serhiy-storchaka** [comment]
Indeed, it is no longer reproduced.

**Pull request closed without merging**